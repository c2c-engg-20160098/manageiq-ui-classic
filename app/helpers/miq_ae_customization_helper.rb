module MiqAeCustomizationHelper
  def dialog_id_action
    url = request.parameters
    if url[:id].present?
      {:id => @record.id.to_s, :action => 'edit'}
    elsif url[:copy].present?
      {:id => url[:copy], :action => 'copy'}
    else
      {:id => '', :action => 'new'}
    end
  end
end
