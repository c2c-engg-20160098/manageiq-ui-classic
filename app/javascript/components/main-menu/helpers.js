export const getHrefByType = (type, href, id) => {
  switch (type) {
    case 'big_iframe':
      return `/dashboard/iframe?id=${id}`;
    case 'modal':
      return undefined;
    default:
      return href;
  }
};

export const getTargetByType = type => (type === 'new_window' ? '_blank' : '_self');

export const getItemId = id => `menu_item_${id}`;

export const getSectionId = id => `menu_section_${id}`;

export const getIdByCategory = (isSection, id) => (isSection ? getSectionId(id) : getItemId(id));

export const handleUnsavedChanges = (type) => {
  if (type === 'modal') {
    return sendDataWithRx({ type: 'showAboutModal' });
  }
  return window.miqCheckForChanges();
};
