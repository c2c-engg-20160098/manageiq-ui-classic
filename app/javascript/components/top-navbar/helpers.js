export const showAboutModal = (e) => {
  e.preventDefault();
  return sendDataWithRx({ type: 'showAboutModal' });
};
