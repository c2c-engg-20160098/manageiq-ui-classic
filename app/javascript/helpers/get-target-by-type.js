const getTargetByType = type => (type === 'new_window' ? '_blank' : '_self');

export default getTargetByType;
