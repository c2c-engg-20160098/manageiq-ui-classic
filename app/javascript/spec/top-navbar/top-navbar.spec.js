import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import RightSection from '../../components/top-navbar/right-section';
import Help from '../../components/top-navbar/help';
import Notifications from '../../components/top-navbar/notifications';
import CustomLogo from '../../components/top-navbar/custom-logo';
import UserOptions from '../../components/top-navbar/user-options';
import '../helpers/sprintf';
import '../helpers/sendDataWithRx';
import '../helpers/miqCheckForChanges';
import '../helpers/miqChangeGroup';


describe('Top navbar tests', () => {
  let customLogo;
  let currentUser;
  let helpMenu;
  let opsExplorerAllowed;
  let applianceName;
  let miqGroups;
  let currentGroup;
  let userMenu;
  let storeUnread;
  let storeNoUnread;
  const mockStore = configureStore();
  const sendDataWithRxSpy = jest.spyOn(window, 'sendDataWithRx');
  const miqCheckForChangesSpy = jest.spyOn(window, 'miqCheckForChanges');
  const miqChangeGroupSpy = jest.spyOn(window, 'miqChangeGroup');


  beforeEach(() => {
    storeUnread = mockStore({
      notificationReducer: {
        unreadCount: 8,
      },
    });
    storeNoUnread = mockStore({
      notificationReducer: {
        unreadCount: 0,
      },
    });
    customLogo = true;
    currentUser = {
      name: 'Administrator',
      userid: 'admin',
    };
    helpMenu = [
      {
        id: 'help',
        title: 'Help',
        type: 'default',
        items: [
          {
            id: 'documentation',
            title: 'Documentation',
            type: 'default',
            items: [],
            visible: true,
            href: '/support/index?support_tab=about',
          },
          {
            id: 'about',
            title: 'About',
            type: 'modal',
            items: [],
            visible: true,
            href: 'javascript:void(0);',
          },
        ],
        visible: true,
        href: '/dashboard/maintab/?tab=help',
      },
    ];
    opsExplorerAllowed = true;
    applianceName = 'EVM';
    miqGroups = [
      {
        id: '10000000000002',
        description: 'EvmGroup-super_administrator',
      },
      {
        id: '10000000000003',
        description: 'EvmGroup-dev',
      },
    ];
    currentGroup = {
      id: '10000000000002',
      description: 'EvmGroup-super_administrator',
    };
    userMenu = [
      {
        id: 'set',
        title: 'User Settings',
        href: null,
        items: [
          {
            id: 'configuration',
            title: 'My Settings',
            href: '/configuration/index',
            items: [],
            visible: true,
          },
          {
            id: 'my_tasks',
            title: 'Tasks',
            href: '/miq_task/index?jobs_tab=tasks',
            items: [],
            visible: true,
          },
        ],
        visible: true,
      },
    ];
  });

  afterEach(() => {
    sendDataWithRxSpy.mockReset();
    miqCheckForChangesSpy.mockReset();
  });

  it('should render correctly', () => {
    const wrapper = mount(
      <Provider store={storeUnread}>
        <RightSection
          customLogo={customLogo}
          currentUser={currentUser}
          helpMenu={helpMenu}
          opsExplorerAllowed={opsExplorerAllowed}
          applianceName={applianceName}
          miqGroups={miqGroups}
          currentGroup={currentGroup}
          userMenu={userMenu}
        />
      </Provider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call sendDataWithRx after click on help menu item with type "modal"', () => {
    const wrapper = mount(
      <Help helpMenu={helpMenu} />,
    );
    wrapper.find('a#help-menu-about').simulate('click');
    expect(sendDataWithRxSpy).toHaveBeenCalledWith({ type: 'showAboutModal' });
  });

  it('should call miqCheckForChanges after click on help menu item with type "default"', () => {
    const wrapper = mount(
      <Help helpMenu={helpMenu} />,
    );
    wrapper.find('a#help-menu-documentation').simulate('click');
    expect(miqCheckForChangesSpy).toHaveBeenCalled();
  });

  it('should call sendDataWithRx after click on notiffication button', () => {
    const wrapper = mount(
      <Provider store={storeUnread}>
        <Notifications />
      </Provider>,
    );
    wrapper.find('a#notifications-btn').simulate('click');
    expect(sendDataWithRxSpy).toHaveBeenCalledWith({ type: 'toggleNotificationsList' });
  });

  it('should not render custom logo when disabled', () => {
    const wrapper = mount(
      <CustomLogo customLogo={false} />,
    );
    expect(wrapper.find('li.dropdown.brand-white-label')).toHaveLength(1);
  });

  it('should not render badge when no unread notifications', () => {
    const wrapper = mount(
      <Provider store={storeNoUnread}>
        <Notifications />
      </Provider>,
    );
    expect(wrapper.find('span.badge.badge-pf-bordered').text()).toEqual('');
  });

  it('should call miqChangeGroup after click on inactive group', () => {
    const wrapper = mount(
      <UserOptions currentUser={currentUser} applianceName={applianceName} miqGroups={miqGroups} currentGroup={currentGroup} userMenu={userMenu} />,
    );
    wrapper.find('a#EvmGroup-dev').simulate('click');
    expect(miqChangeGroupSpy).toHaveBeenCalled();
  });

  it('should render disabled item in case of one group', () => {
    const wrapper = mount(
      <UserOptions currentUser={currentUser} applianceName={applianceName} miqGroups={[miqGroups[0]]} currentGroup={currentGroup} userMenu={userMenu} />,
    );
    expect(wrapper.find('a#single-group-item')).toHaveLength(1);
  });

  it('should call miqCheckForChanges after click on user menu item', () => {
    const wrapper = mount(
      <UserOptions currentUser={currentUser} applianceName={applianceName} miqGroups={miqGroups} currentGroup={currentGroup} userMenu={userMenu} />,
    );
    wrapper.find('a#user-menu-tasks').simulate('click');
    expect(miqCheckForChangesSpy).toHaveBeenCalled();
  });

  it('should call miqCheckForChanges before logout', () => {
    const wrapper = mount(
      <UserOptions currentUser={currentUser} applianceName={applianceName} miqGroups={miqGroups} currentGroup={currentGroup} userMenu={userMenu} />,
    );
    wrapper.find('a#logout-btn').simulate('click');
    expect(miqCheckForChangesSpy).toHaveBeenCalled();
  });
});
