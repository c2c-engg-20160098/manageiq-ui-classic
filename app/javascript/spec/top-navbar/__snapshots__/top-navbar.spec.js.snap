// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top navbar tests should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <RightSection
    applianceName="EVM"
    currentGroup={
      Object {
        "description": "EvmGroup-super_administrator",
        "id": "10000000000002",
      }
    }
    currentUser={
      Object {
        "name": "Administrator",
        "userid": "admin",
      }
    }
    customLogo={true}
    helpMenu={
      Array [
        Object {
          "href": "/dashboard/maintab/?tab=help",
          "id": "help",
          "items": Array [
            Object {
              "href": "/support/index?support_tab=about",
              "id": "documentation",
              "items": Array [],
              "title": "Documentation",
              "type": "default",
              "visible": true,
            },
            Object {
              "href": "javascript:void(0);",
              "id": "about",
              "items": Array [],
              "title": "About",
              "type": "modal",
              "visible": true,
            },
          ],
          "title": "Help",
          "type": "default",
          "visible": true,
        },
      ]
    }
    miqGroups={
      Array [
        Object {
          "description": "EvmGroup-super_administrator",
          "id": "10000000000002",
        },
        Object {
          "description": "EvmGroup-dev",
          "id": "10000000000003",
        },
      ]
    }
    opsExplorerAllowed={true}
    userMenu={
      Array [
        Object {
          "href": null,
          "id": "set",
          "items": Array [
            Object {
              "href": "/configuration/index",
              "id": "configuration",
              "items": Array [],
              "title": "My Settings",
              "visible": true,
            },
            Object {
              "href": "/miq_task/index?jobs_tab=tasks",
              "id": "my_tasks",
              "items": Array [],
              "title": "Tasks",
              "visible": true,
            },
          ],
          "title": "User Settings",
          "visible": true,
        },
      ]
    }
  >
    <CustomLogo
      customLogo={true}
    >
      <li
        className="dropdown brand-white-label whitelabeled"
      />
    </CustomLogo>
    <Connect(Notifications)>
      <Notifications
        dispatch={[Function]}
        unreadCount={8}
      >
        <li
          className="dropdown"
        >
          <a
            className="nav-item-iconic drawer-pf-trigger-icon"
            id="notifications-btn"
            onClick={[Function]}
            title="8 unread notifications"
          >
            <Icon
              name="bell"
              type="fa"
            >
              <FontAwesome
                name="bell"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-bell"
                />
              </FontAwesome>
            </Icon>
            <span
              className="badge badge-pf-bordered"
            >
               
            </span>
          </a>
        </li>
      </Notifications>
    </Connect(Notifications)>
    <Help
      helpMenu={
        Array [
          Object {
            "href": "/dashboard/maintab/?tab=help",
            "id": "help",
            "items": Array [
              Object {
                "href": "/support/index?support_tab=about",
                "id": "documentation",
                "items": Array [],
                "title": "Documentation",
                "type": "default",
                "visible": true,
              },
              Object {
                "href": "javascript:void(0);",
                "id": "about",
                "items": Array [],
                "title": "About",
                "type": "modal",
                "visible": true,
              },
            ],
            "title": "Help",
            "type": "default",
            "visible": true,
          },
        ]
      }
    >
      <ForwardRef
        componentClass={[Function]}
        id="help-dd"
        key="help"
      >
        <Uncontrolled(Dropdown)
          componentClass={[Function]}
          id="help-dd"
          innerRef={null}
        >
          <Dropdown
            bsClass="dropdown"
            componentClass={[Function]}
            id="help-dd"
            onToggle={[Function]}
          >
            <componentClass
              className="dropdown"
            >
              <li
                className="dropdown"
              >
                <DropdownToggle
                  bsClass="dropdown-toggle"
                  bsRole="toggle"
                  componentClass={[Function]}
                  id="help-dd"
                  key=".0"
                  noCaret={true}
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  open={false}
                  useAnchor={false}
                >
                  <Button
                    active={false}
                    aria-expanded={false}
                    aria-haspopup={true}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="dropdown-toggle"
                    componentClass={[Function]}
                    disabled={false}
                    id="help-dd"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                  >
                    <componentClass
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="dropdown-toggle btn btn-default"
                      disabled={false}
                      id="help-dd"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      type="button"
                    >
                      <a
                        aria-expanded={false}
                        aria-haspopup={true}
                        className="dropdown-toggle nav-item-iconic"
                        data-toggle="dropdown"
                        disabled={false}
                        id="help-menu"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        type="button"
                      >
                        <Icon
                          name="help"
                          title="Help"
                          type="pf"
                        >
                          <PatternflyIcon
                            className=""
                            name="help"
                            title="Help"
                          >
                            <span
                              aria-hidden="true"
                              className="pficon pficon-help"
                              title="Help"
                            />
                          </PatternflyIcon>
                        </Icon>
                      </a>
                    </componentClass>
                  </Button>
                </DropdownToggle>
                <DropdownMenu
                  aria-labelledby="help-menu"
                  bsClass="dropdown-menu"
                  bsRole="menu"
                  key=".1"
                  labelledBy="help-dd"
                  onClose={[Function]}
                  onSelect={[Function]}
                  pullRight={false}
                >
                  <RootCloseWrapper
                    disabled={true}
                    event="click"
                    onRootClose={[Function]}
                  >
                    <ul
                      aria-labelledby="help-dd"
                      className="dropdown-menu"
                      role="menu"
                    >
                      <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        header={false}
                        href="/support/index?support_tab=about"
                        id="help-menu-documentation"
                        key=".$documentation"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        target="_self"
                      >
                        <li
                          className=""
                          role="presentation"
                        >
                          <SafeAnchor
                            componentClass="a"
                            href="/support/index?support_tab=about"
                            id="help-menu-documentation"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                            target="_self"
                          >
                            <a
                              href="/support/index?support_tab=about"
                              id="help-menu-documentation"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                              target="_self"
                            >
                              Documentation
                            </a>
                          </SafeAnchor>
                        </li>
                      </MenuItem>
                      <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        header={false}
                        href=""
                        id="help-menu-about"
                        key=".$about"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        target="_self"
                      >
                        <li
                          className=""
                          role="presentation"
                        >
                          <SafeAnchor
                            componentClass="a"
                            href=""
                            id="help-menu-about"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                            target="_self"
                          >
                            <a
                              href="#"
                              id="help-menu-about"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                              target="_self"
                            >
                              About
                            </a>
                          </SafeAnchor>
                        </li>
                      </MenuItem>
                    </ul>
                  </RootCloseWrapper>
                </DropdownMenu>
              </li>
            </componentClass>
          </Dropdown>
        </Uncontrolled(Dropdown)>
      </ForwardRef>
    </Help>
    <Configuration
      opsExplorerAllowed={true}
    >
      <li
        className="dropdown"
      >
        <a
          className="nav-item-iconic"
          href="/ops/explorer"
          title="Configuration"
        >
          <i
            className="fa fa-cog"
          />
        </a>
      </li>
    </Configuration>
    <UserOptions
      applianceName="EVM"
      currentGroup={
        Object {
          "description": "EvmGroup-super_administrator",
          "id": "10000000000002",
        }
      }
      currentUser={
        Object {
          "name": "Administrator",
          "userid": "admin",
        }
      }
      miqGroups={
        Array [
          Object {
            "description": "EvmGroup-super_administrator",
            "id": "10000000000002",
          },
          Object {
            "description": "EvmGroup-dev",
            "id": "10000000000003",
          },
        ]
      }
      userMenu={
        Array [
          Object {
            "href": null,
            "id": "set",
            "items": Array [
              Object {
                "href": "/configuration/index",
                "id": "configuration",
                "items": Array [],
                "title": "My Settings",
                "visible": true,
              },
              Object {
                "href": "/miq_task/index?jobs_tab=tasks",
                "id": "my_tasks",
                "items": Array [],
                "title": "Tasks",
                "visible": true,
              },
            ],
            "title": "User Settings",
            "visible": true,
          },
        ]
      }
    >
      <ForwardRef
        componentClass={[Function]}
        id="dropdownMenu2"
      >
        <Uncontrolled(Dropdown)
          componentClass={[Function]}
          id="dropdownMenu2"
          innerRef={null}
        >
          <Dropdown
            bsClass="dropdown"
            componentClass={[Function]}
            id="dropdownMenu2"
            onToggle={[Function]}
          >
            <componentClass
              className="dropdown"
            >
              <li
                className="dropdown"
              >
                <DropdownToggle
                  bsClass="dropdown-toggle"
                  bsRole="toggle"
                  componentClass={[Function]}
                  id="dropdownMenu2"
                  key=".0"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  open={false}
                  useAnchor={false}
                >
                  <Button
                    active={false}
                    aria-expanded={false}
                    aria-haspopup={true}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="dropdown-toggle"
                    componentClass={[Function]}
                    disabled={false}
                    id="dropdownMenu2"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                  >
                    <componentClass
                      aria-expanded={false}
                      aria-haspopup={true}
                      className="dropdown-toggle btn btn-default"
                      disabled={false}
                      id="dropdownMenu2"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                      type="button"
                    >
                      <a
                        aria-expanded={false}
                        aria-haspopup="true"
                        className="dropdown-toggle nav-item-iconic"
                        data-toggle="dropdown"
                        disabled={false}
                        id="dropdownMenu2"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        type="button"
                      >
                        <Icon
                          name="user"
                          type="pf"
                        >
                          <PatternflyIcon
                            className=""
                            name="user"
                          >
                            <span
                              aria-hidden="true"
                              className="pficon pficon-user"
                            />
                          </PatternflyIcon>
                        </Icon>
                        <p
                          className="navbar__user-name"
                          data-userid="admin"
                          id="username_display"
                        >
                          Administrator | EVM
                        </p>
                         
                        <span
                          className="caret"
                        />
                      </a>
                    </componentClass>
                  </Button>
                </DropdownToggle>
                <DropdownMenu
                  bsClass="dropdown-menu"
                  bsRole="menu"
                  key=".1"
                  labelledBy="dropdownMenu2"
                  onClose={[Function]}
                  onSelect={[Function]}
                  pullRight={false}
                >
                  <RootCloseWrapper
                    disabled={true}
                    event="click"
                    onRootClose={[Function]}
                  >
                    <ul
                      aria-labelledby="dropdownMenu2"
                      className="dropdown-menu"
                      role="menu"
                    >
                      <MenuItem
                        bsClass="dropdown"
                        disabled={true}
                        divider={false}
                        header={false}
                        key=".0"
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                      >
                        <li
                          className="disabled"
                          role="presentation"
                        >
                          <SafeAnchor
                            componentClass="a"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            <a
                              href="#"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                            >
                              Server: EVM
                            </a>
                          </SafeAnchor>
                        </li>
                      </MenuItem>
                      <li
                        className="dropdown-submenu pull-left"
                        id="droptownMenuGroup"
                        key=".1"
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                      >
                        <DropdownToggle
                          bsClass="dropdown-toggle"
                          bsRole="toggle"
                          componentClass={[Function]}
                          noCaret={true}
                          open={false}
                          useAnchor={true}
                        >
                          <SafeAnchor
                            aria-expanded={false}
                            aria-haspopup={true}
                            className="dropdown-toggle"
                            componentClass={[Function]}
                            role="button"
                          >
                            <componentClass
                              aria-expanded={false}
                              aria-haspopup={true}
                              className="dropdown-toggle"
                              href="#"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="button"
                            >
                              <a
                                href="#"
                              >
                                Change Group
                              </a>
                            </componentClass>
                          </SafeAnchor>
                        </DropdownToggle>
                        <DropdownMenu
                          bsClass="dropdown-menu"
                          bsRole="menu"
                          className="scrollable-menu"
                          pullRight={false}
                        >
                          <RootCloseWrapper
                            disabled={true}
                            event="click"
                            onRootClose={[Function]}
                          >
                            <ul
                              className="scrollable-menu dropdown-menu"
                              role="menu"
                            >
                              <MenuItem
                                bsClass="dropdown"
                                disabled={true}
                                divider={false}
                                header={false}
                                id="current-group"
                                key=".$10000000000002"
                                onKeyDown={[Function]}
                                onSelect={null}
                                title="Currently Selected Group"
                              >
                                <li
                                  className="disabled"
                                  role="presentation"
                                >
                                  <SafeAnchor
                                    componentClass="a"
                                    id="current-group"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    role="menuitem"
                                    tabIndex="-1"
                                    title="Currently Selected Group"
                                  >
                                    <a
                                      href="#"
                                      id="current-group"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      role="menuitem"
                                      tabIndex="-1"
                                      title="Currently Selected Group"
                                    >
                                      EvmGroup-super_administrator (Current Group)
                                    </a>
                                  </SafeAnchor>
                                </li>
                              </MenuItem>
                              <MenuItem
                                bsClass="dropdown"
                                disabled={false}
                                divider={false}
                                header={false}
                                id="EvmGroup-dev"
                                key=".$10000000000003"
                                onClick={[Function]}
                                onKeyDown={[Function]}
                                onSelect={null}
                                title="Change to this Group"
                              >
                                <li
                                  className=""
                                  role="presentation"
                                >
                                  <SafeAnchor
                                    componentClass="a"
                                    id="EvmGroup-dev"
                                    onClick={[Function]}
                                    onKeyDown={[Function]}
                                    role="menuitem"
                                    tabIndex="-1"
                                    title="Change to this Group"
                                  >
                                    <a
                                      href="#"
                                      id="EvmGroup-dev"
                                      onClick={[Function]}
                                      onKeyDown={[Function]}
                                      role="menuitem"
                                      tabIndex="-1"
                                      title="Change to this Group"
                                    >
                                      EvmGroup-dev
                                    </a>
                                  </SafeAnchor>
                                </li>
                              </MenuItem>
                            </ul>
                          </RootCloseWrapper>
                        </DropdownMenu>
                      </li>
                      <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={true}
                        header={false}
                      >
                        <li
                          className="divider"
                          role="separator"
                        />
                      </MenuItem>
                      <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        header={false}
                        href="/configuration/index"
                        id="user-menu-my settings"
                        key="configuration"
                        onClick={[Function]}
                      >
                        <li
                          className=""
                          role="presentation"
                        >
                          <SafeAnchor
                            componentClass="a"
                            href="/configuration/index"
                            id="user-menu-my settings"
                            onClick={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            <a
                              href="/configuration/index"
                              id="user-menu-my settings"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                            >
                              My Settings
                            </a>
                          </SafeAnchor>
                        </li>
                      </MenuItem>
                      <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        header={false}
                        href="/miq_task/index?jobs_tab=tasks"
                        id="user-menu-tasks"
                        key="my_tasks"
                        onClick={[Function]}
                      >
                        <li
                          className=""
                          role="presentation"
                        >
                          <SafeAnchor
                            componentClass="a"
                            href="/miq_task/index?jobs_tab=tasks"
                            id="user-menu-tasks"
                            onClick={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                          >
                            <a
                              href="/miq_task/index?jobs_tab=tasks"
                              id="user-menu-tasks"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                            >
                              Tasks
                            </a>
                          </SafeAnchor>
                        </li>
                      </MenuItem>
                      <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={true}
                        header={false}
                        key=".3"
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                      >
                        <li
                          className="divider"
                          onKeyDown={[Function]}
                          role="separator"
                        />
                      </MenuItem>
                      <MenuItem
                        bsClass="dropdown"
                        disabled={false}
                        divider={false}
                        header={false}
                        href="/dashboard/logout"
                        id="logout-btn"
                        key=".4"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        onSelect={[Function]}
                        title="Click to Logout"
                      >
                        <li
                          className=""
                          role="presentation"
                        >
                          <SafeAnchor
                            componentClass="a"
                            href="/dashboard/logout"
                            id="logout-btn"
                            onClick={[Function]}
                            onKeyDown={[Function]}
                            role="menuitem"
                            tabIndex="-1"
                            title="Click to Logout"
                          >
                            <a
                              href="/dashboard/logout"
                              id="logout-btn"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                              title="Click to Logout"
                            >
                              Logout
                            </a>
                          </SafeAnchor>
                        </li>
                      </MenuItem>
                    </ul>
                  </RootCloseWrapper>
                </DropdownMenu>
              </li>
            </componentClass>
          </Dropdown>
        </Uncontrolled(Dropdown)>
      </ForwardRef>
    </UserOptions>
  </RightSection>
</Provider>
`;
